homeassistant/components/osoenergy/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for OSO Energy binary sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from apyosoenergyapi import OSOEnergy
+from apyosoenergyapi.helper.const import OSOEnergyBinarySensorData
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import OSOEnergyEntity
+from .const import DOMAIN
+
+
+@dataclass(frozen=True, kw_only=True)
+class OSOEnergyBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Class describing OSO Energy heater binary sensor entities."""
+
+    value_fn: Callable[[OSOEnergy], bool]
+
+
+SENSOR_TYPES: dict[str, OSOEnergyBinarySensorEntityDescription] = {
+    "power_save": OSOEnergyBinarySensorEntityDescription(
+        key="power_save",
+        translation_key="power_save",
+        value_fn=lambda entity_data: entity_data.state,
+    ),
+    "extra_energy": OSOEnergyBinarySensorEntityDescription(
+        key="extra_energy",
+        translation_key="extra_energy",
```

Comment:
Can we control the extra energy somehow? In my opinion, this sounds like a switch, where I can enable/disable the extra energy.

Reply:
The extra energy and sleep mode can be controlled, but not as a switch. The enabling of this requires additional parameters like period of time and will be introduced as scripts in a separate pull request.